api/src/client/ApiFormationClient.test.ts

```diff
@@ -1223,6 +1223,514 @@ describe("ApiFormationClient", () => {
       });
     });
 
+    describe("when Nonprofit", () => {
+      it("posts to the endpoint with the api formation object – when nonprofit provisions are IN_BYLAWS", async () => {
+        const stubResponse = generateApiResponse({});
+        mockAxios.post.mockResolvedValue({ data: stubResponse });
+        const incorporators = [
+          generateFormationIncorporator({
+            name: "faraz",
+            signature: true,
+            title: "General Partner",
+            ...generateFormationNJAddress({}),
+          }),
+        ];
+
+        const members = [
+          generateFormationMember({}),
+          generateFormationMember({}),
+          generateFormationMember({}),
+        ];
+        const formationFormData = generateFormationFormData(
+          {
+            agentNumberOrManual: "MANUAL_ENTRY",
+            hasNonprofitBoardMembers: true,
+            nonprofitBoardMemberQualificationsSpecified: "IN_BYLAWS",
+            nonprofitBoardMemberQualificationsTerms: "LOREM IPSUM DOLOR SIT AMET",
+            nonprofitBoardMemberRightsSpecified: "IN_BYLAWS",
+            nonprofitBoardMemberRightsTerms: "LOREM IPSUM DOLOR SIT AMET",
+            nonprofitTrusteesMethodSpecified: "IN_BYLAWS",
+            nonprofitTrusteesMethodTerms: "LOREM IPSUM DOLOR SIT AMET",
+            nonprofitAssetDistributionSpecified: "IN_BYLAWS",
+            nonprofitAssetDistributionTerms: "LOREM IPSUM DOLOR SIT AMET",
```

Comment:
[pebble] I would remove all these LOREM IPSUMs because (if I understand correctly) when the user selects "IN_BYLAWS" then they are NOT given a text box and an opportunity to type text. So this isn't really a state that is possible?

Reply:
That's correct, and testing around that _shouldn't_ allow this to happen. I helped AnJu write this one. I was thinking that by including this we would be verifying that the `ApiFormationClient` wouldn't include this information even if it was supplied. Passing the fields in this state to the NICUSA API will return an error.

@aloverso, would you test for that case? Or would you be confident enough in other testing that this state of the data wouldn't be possible?

Reply:
@jphechter That's a good point, but I think you actually do get to test that behavior "for free" by removing them here. Because this chunk of code is part of the overrides being passed to `generateFormationFormData` - anything not overridden will get a default random string generated for it.

So if these LOREM IPSUMs get removed from the overrides in the test setup here, they'll actually still have values in the object that gets operated in on the ApiClient, so you're still implicitly testing that it is not included.

My rationale for removing them is that they clog up your visual filter of someone skimming and understanding the code. It seems to me the important part of this test is that it's saying what happens when these fields are IN_BYLAWS. So making that part of the setup stand alone without adding unnecessary strings that don't affect its outcome makes the test tighter overall. imo